daemon/src/taker_cfd.rs

```diff
@@ -169,6 +171,17 @@ where
         let order_to_take = order_to_take.context("Order to take could not be found in current maker offers, you might have an outdated offer")?;
         self.current_maker_offers.replace(maker_offers);
 
+        // If the offer is not safe to take we return and error and the offer is removed (by taking
+        // it, see code above), so on the first take of an outdated offer we print the error and
+        // then the offer for the taker will be gone until a new offer arrives from the
+        // maker
+        if !order_to_take.is_safe_to_take(OffsetDateTime::now_utc()) {
```

Comment:
🔧 This could be a bit clearer. Particularly since you don't need to refer back to the "code above".

I don't think the code below needs a comment because it reads nicely: "if the order to take is not safe to take we refuse to take it because it appears to be outdated".

```suggestion
        // the offer we are instructed to take is removed from the
        // set of available offers immediately so that we don't attempt
        // to take it more than once
        self.current_maker_offers.replace(maker_offers);

        if !order_to_take.is_safe_to_take(OffsetDateTime::now_utc()) {
```

Reply:
fair enough :)